### PR TITLE
Align to consensus release v1.4.0-beta.2

### PR DESCRIPTION
**Pull request by terencechain**
Todos:
- [x] https://github.com/ethereum/consensus-specs/pull/3352
- [x] https://eips.ethereum.org/EIPS/eip-7514
- [ ] https://github.com/ethereum/consensus-specs/pull/3463
- [x] https://github.com/ethereum/consensus-specs/pull/3484

### CODE REVIEW

File: testing/endtoend/evaluators/operations.go [hunk 1]
```diff
@@ -29,8 +29,6 @@ import (
 	"google.golang.org/protobuf/types/known/emptypb"
 )
 
-// churnLimit is normally 4 unless the validator set is extremely large.
-var churnLimit = 4
```
Comment (terencechain) [inline testing/endtoend/evaluators/operations.go:33]: I am removing this to use the default config

File: config/params/mainnet_config.go [hunk 2]
```diff
@@ -270,6 +270,8 @@ var mainnetBeaconConfig = &BeaconChainConfig{
 
 	// Subnet value
 	BlobsidecarSubnetCount: 6,
+
+	MaxPerEpochActivationChurnLimit: 8,
```
Comment (james-prysm) [inline config/params/mainnet_config.go:274]: do we not need the min here too?

File: config/params/testnet_e2e_config.go [hunk 3]
```diff
@@ -96,6 +96,9 @@ func E2EMainnetTestConfig() *BeaconChainConfig {
 	e2eConfig.CapellaForkVersion = []byte{3, 0, 0, 254}
 	e2eConfig.DenebForkVersion = []byte{4, 0, 0, 254}
 
+	// Deneb changes.
+	e2eConfig.MinPerEpochChurnLimit = 2
```
Comment (james-prysm) [inline config/params/testnet_e2e_config.go:100]: do we not need max here?

File: beacon-chain/core/epoch/epoch_processing.go [hunk 4]
```diff
@@ -146,6 +146,10 @@ func ProcessRegistryUpdates(ctx context.Context, state state.BeaconState) (state
 	if churnLimit < limit {
 		limit = churnLimit
 	}
+	// Cap churn limit to max per epoch churn limit. New in EIP7514.
+	if limit > params.BeaconConfig().MaxPerEpochActivationChurnLimit {
```
Comment (james-prysm) [inline beacon-chain/core/epoch/epoch_processing.go:150]: where is the minimum set for this

File: config/params/mainnet_config.go (hunk 2 above)
Comment (terencechain) [reply config/params/mainnet_config.go:274]: there's no change here

File: config/params/testnet_e2e_config.go (hunk 3 above)
Comment (terencechain) [reply config/params/testnet_e2e_config.go:100]: It should just use the default which is 8

File: beacon-chain/core/epoch/epoch_processing.go (hunk 4 above)
Comment (terencechain) [reply beacon-chain/core/epoch/epoch_processing.go:150]: Minimal is used in other places. It's not part of the change besides the value

File: testing/endtoend/evaluators/operations.go [hunk 5]
```diff
@@ -249,8 +247,8 @@ func activatesDepositedValidators(ec *e2etypes.EvaluationContext, conns ...*grpc
 		return fmt.Errorf("missing %d validators for post-genesis deposits", len(expected))
 	}
 
-	if deposits != churnLimit {
-		return fmt.Errorf("expected %d deposits to be processed in epoch %d, received %d", churnLimit, epoch, deposits)
+	if deposits != int(params.BeaconConfig().MinPerEpochChurnLimit) {
```
Comment (rauljordan) [inline testing/endtoend/evaluators/operations.go:250]: prefer to cast deposits to uint

**Review by rauljordan — DISMISSED** [review]

**Pull request closed without merging**